### ISSUE TITLE
[CRITICAL] Unbounded Memory Growth in Layer Cache - OOM Risk

### ISSUE THREAD

**Issue by user1303836** [pxxxe/azu#7]
## Summary

The layer loader maintains an in-memory cache of loaded layers with **no size limit or eviction policy**. This will cause workers to run out of VRAM/RAM as they process different models.

## Affected Files

- `client/worker/layer_loader.py` (lines 9-13)
- `client/worker/main.py` (layer dictionaries)

## Issue Details

```python
class LayerLoader:
    def __init__(self, registry_url, cache_dir="./layer_cache"):
        self.registry_url = registry_url
        self.cache_dir = Path(cache_dir)
        self.cache_dir.mkdir(exist_ok=True, parents=True)
        self.loaded_cache = {}  # NO SIZE LIMIT - grows forever
```

Similarly in the worker:
```python
self.dense_layers: Dict[int, nn.Module] = {}
self.moe_routers: Dict[int, nn.Module] = {}
self.moe_experts: Dict[Tuple[int, int], nn.Module] = {}
```

### Problem Scenario

1. Worker processes model A with 80 layers → caches all 80 layers
2. Worker processes model B with 70 layers → caches 70 more layers
3. Worker processes model C → runs out of VRAM, crashes
4. No warning before OOM, no graceful degradation

### Impact

1. **Worker Crashes**: OOM kills without warning
2. **Silent Degradation**: VRAM exhaustion causes slowdowns before crash
3. **Unpredictable Behavior**: Random failures depending on job history
4. **No Recovery**: Worker must be restarted manually

## Recommended Fix

1. Implement LRU (Least Recently Used) eviction policy:
   ```python
   from functools import lru_cache
   # Or use custom cache with VRAM awareness
   ```

2. Track actual VRAM usage with `torch.cuda.memory_allocated()`

3. Add configurable max cache size:
   ```python
   MAX_CACHED_LAYERS = int(os.getenv('MAX_CACHED_LAYERS', 50))
   ```

4. Evict oldest layers when approaching VRAM limit

5. Report cache status to scheduler so it can make informed placement decisions

## Severity

**CRITICAL** - Workers will crash unpredictably in production.
